pkg/parse/run.go

```diff
@@ -491,6 +491,7 @@ func parseAndUpdate(ctx context.Context, p Parser, trigger string, state *reconc
 	klog.V(3).Info("Parser starting...")
 	sourceErrs := parseSource(ctx, p, trigger, state)
 	klog.V(3).Info("Parser stopped")
+	oldCommit := state.sourceStatus.commit
```

Comment:
Does this get set from the RSync status or will it be the empty string when the reconciler first starts up?

Reply:
I can add a check for empty string when resetting, are you indicating that the reset when reconciler first started might create a new stream with empty commit?

Reply:
Yeah. If it's empty string, we don't know what the previous commit was and can't reset it. We should probably skip sending the reset metric in this case. But it leaves open a gap when if the commit changes when the reconciler is down the metric won't be reset. This could easily happy during upgrade or rescheduling, especially if the repo moves quickly or scheduling is slow (like autopilot) or broken for some other reason.

Reply:
However, if `state.sourceStatus.commit` is populated from the RSync status before this, then it should work ok. As long as we skip metric reset when the oldCommit is empty.

Reply:
Breaking down by case:
- A new commit comes in when reconciler down during upgrade - no metric would be sent anyways, and the previous streams will be killed due to the pod restart
- Got oldCommit as empty string - not likely, but maybe when the reconciler first starts up and performing first sync? I added a check for empty string during reset.

Reply:
The state is updated in line 282 before running ParseAndUpdate where the fix takes place, if that's where the sourceStatus gets updated from Rsync state then the oldCommit should be legit?